tests/integration/test_database.py

```diff
@@ -82,6 +90,131 @@ async def test_build_and_deploy(ops_test: OpsTest):
 @pytest.mark.order(2)
 @pytest.mark.abort_on_fail
 @pytest.mark.database_tests
+async def test_password_rotation(ops_test: OpsTest):
+    """Rotate password and confirm changes."""
+    num_units = len(ops_test.model.applications[DATABASE_APP_NAME].units)
+    random_unit = ops_test.model.applications[DATABASE_APP_NAME].units[num_units - 1]
```

Comment:
you could potentially index the units array by `-1` and achieve the same - but is more readable in my opinion